samples/transform_feedback_separated_2.html

```diff
@@ -0,0 +1,405 @@
+            gl.vertexAttribPointer(SPAWNTIME_LOCATION, 1, gl.FLOAT, false, 0, 0);
+
+            gl.bindBuffer(gl.ARRAY_BUFFER, vboArray[Particle.LIFE_TIME]);
+            gl.vertexAttribPointer(LIFETIME_LOCATION, 1, gl.FLOAT, false, 0, 0);
+
+            gl.bindBuffer(gl.ARRAY_BUFFER, vboArray[Particle.ID]);
+            gl.vertexAttribPointer(ID_LOCATION, 1, gl.FLOAT, false, 0, 0);
+
+            gl.enableVertexAttribArray(POSITION_LOCATION);
+            gl.enableVertexAttribArray(VELOCITY_LOCATION);
+            gl.enableVertexAttribArray(SPAWNTIME_LOCATION);
+            gl.enableVertexAttribArray(LIFETIME_LOCATION);
+            gl.enableVertexAttribArray(ID_LOCATION);
+        }
+
+        function emitParticles() {
+            time = Date.now();
+
+            var programEmit = programs[PROGRAM_EMIT];
+
+            // Toggle source and destination VBO
+            var sourceVBO = particleVBOs[currentSourceIdx];
+            var sourceVAO = particleVAOs[currentSourceIdx];
+
+            var destinationVBO = particleVBOs[(currentSourceIdx + 1) % 2];
+
+            gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, transformFeedback);
+
+            gl.useProgram(programEmit);
+
+            setupVertexAttributes(sourceVAO, sourceVBO);
+
+            // Set transform feedback buffer
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, destinationVBO[Particle.POSITION]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 1, destinationVBO[Particle.VELOCITY]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 2, destinationVBO[Particle.SPAWN_TIME]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 3, destinationVBO[Particle.LIFE_TIME]);
+
+            // Turn off rasterization - we are not drawing
+            gl.enable(gl.RASTERIZER_DISCARD);
+
+            // Set uniforms
+            gl.uniform1f(emitTimeLocation, time);
+            gl.uniform2f(emitAccelerationLocation, 0.0, ACCELERATION);
+
+            // Emit particles using transform feedback
+            gl.beginTransformFeedback(gl.POINTS);
+            gl.drawArrays(gl.POINTS, 0, NUM_PARTICLES);
+            gl.endTransformFeedback();
+
+            // Restore state
+            gl.disable(gl.RASTERIZER_DISCARD);
+            gl.useProgram(null);
+            gl.bindBuffer(gl.ARRAY_BUFFER, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 1, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 2, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 3, null);
+            gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null);
+
+            // Ping pong the buffers
+            currentSourceIdx = (currentSourceIdx + 1) % 2;
+        }
+
+        function render() {
+            // Spawn particles
+            emitParticles();
+
+            time = Date.now();

```

Comment:
It probably doesn't matter for this app, but a more robust way to do this is to get `Date.now()` when the app first starts, then pass `Date.now() - that_variable` as the uniform to avoid the giant numbers.